library/unwind/src/libunwind.rs

```diff
@@ -77,9 +77,16 @@ pub enum _Unwind_Context {}
 
 pub type _Unwind_Exception_Cleanup_Fn =
     extern "C" fn(unwind_code: _Unwind_Reason_Code, exception: *mut _Unwind_Exception);
+
+// These link attrs is to ensure libstd.so export those function
```

Comment:
1. What effect does `#[link]` attribute have symbol exports? Do I understand correctly that link attributes cannot be centralized in `library/unwind/src/lib.rs` due to that effect?
2. How is this related to the "This libunwind.a is built without hiding its symbol" logic from `src/bootstrap/compile.rs`?

Reply:
1. Yes, those `#[link]` attributes export those symbols to libstd.so. They are added to fix https://github.com/rust-lang/rust/issues/62088. Those symbols is needed by llvm, and empty `extern "C" {}` don't export those symbols.  It's only needed when build `rustc` with feature `llvm-libunwind` and `static-libstdcpp`
2. Building rustc with feature `llvm-libunwind` is broken again because https://github.com/rust-lang/rust/pull/72746.  The possible way to fix this is mentioned in https://github.com/rust-lang/rust/issues/76020#issuecomment-707528177

Reply:
>Yes, those `#[link]` attributes export those symbols to libstd.so.

Using `#[link]` for performing an entirely unrelated task is a really inappropriate behavior from rustc, although I can't say I'm surprised at this point.
I hope this is something that can be rectified in the future.

Reply:
Could you make the comment more explicit/verbose like "Using these `#[link]` attributes on the extern block triggers some questionable logic in rustc that marks symbols declared in such block as exported, so the functions below become visible from `libstd.so` at link time, which is important on targets without a system unwinder (see #62088)".

Reply:
The comment is added. And the magic logic is at https://github.com/rust-lang/rust/blob/master/compiler/rustc_codegen_ssa/src/back/symbol_export.rs#L81 which is introduced in https://github.com/rust-lang/rust/pull/48611